pkg/virtualKubelet/translation/serviceEnv/serviceEnv.go

```diff
@@ -0,0 +1,120 @@
+package serviceEnv
+
+import (
+	apimgmgt "github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/storage"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"k8s.io/kubernetes/pkg/kubelet/envvars"
+)
+
+var masterServices = sets.NewString("kubernetes")
+
+func TranslateServiceEnvVariables(pod *v1.Pod, localNS string, nattedNS string, cacheManager storage.CacheManagerReader) (*v1.Pod, error) {
+	resPod := pod.DeepCopy()
+
+	enableServiceLinks := v1.DefaultEnableServiceLinks
+	if resPod.Spec.EnableServiceLinks != nil {
+		enableServiceLinks = *resPod.Spec.EnableServiceLinks
+	}
+
+	envs, err := getServiceEnvVarMap(localNS, enableServiceLinks, nattedNS, cacheManager)
+	if err != nil {
+		return nil, err
+	}
+	for i, container := range resPod.Spec.Containers {
+		resPod.Spec.Containers[i] = *setInContainer(envs, container)
+	}
+	for i, container := range resPod.Spec.InitContainers {
+		resPod.Spec.InitContainers[i] = *setInContainer(envs, container)
+	}
+	return resPod, nil
+}
+
+func setInContainer(envs map[string]string, container v1.Container) *v1.Container {
+	for k, v := range envs {
+		found := false
+		for _, env := range container.Env {
+			if env.Name == k {
+				found = true
+				break
+			}
+		}
+		if !found {
+			container.Env = append(container.Env, v1.EnvVar{
+				Name:  k,
+				Value: v,
+			})
+		}
+	}
+	return container.DeepCopy()
+}
+
+func getServiceEnvVarMap(ns string, enableServiceLinks bool, remoteNs string, cacheManager storage.CacheManagerReader) (map[string]string, error) {
+	var (
+		serviceMap = make(map[string]*v1.Service)
+		m          = make(map[string]string)
+	)
+
+	// search for services in the same namespaces of the pod
+	services, err := cacheManager.ListHomeNamespacedObject(apimgmgt.Services, ns)
+	if err != nil {
+		return nil, err
+	}
+
+	// project the services in namespace ns onto the master services
+	for i := range services {
+		// We always want to add environment variables for master kubernetes service
+		// from the default namespace, even if enableServiceLinks is false.
+		// We also add environment variables for other services in the same
+		// namespace, if enableServiceLinks is true.
+
+		tmp := services[i]
+		service, ok := tmp.(*v1.Service)
+		if !ok {
+			continue
+		}
+		serviceName := service.Name
+
+		if service.Namespace == metav1.NamespaceDefault && masterServices.Has(serviceName) {
+			// TODO: we have to set env vars for our home kubernetes service
```

Comment:
I am not sure that we should set this environment variable for now (I would remove this if statement with its TODO and maybe we can keep track of this in an issue). In order to allow the remote pods communicating with the home API server in this way, we also should reflect that namespace and nat its address, but for now, it is blacklisted (i.e., not reflected [here](https://github.com/liqotech/liqo/blob/734d7544da18c8af58d9aae44df37167965b80a0/pkg/virtualKubelet/apiReflection/reflectors/outgoing/blacklists.go#L14))

Reply:
the Kubernetes service translation is already done in another pr [here](https://github.com/liqotech/liqo/pull/362/files), and it is done not with a service but by providing the IP address of the home API server

Reply:
ok, then I suppose we can remove this condition related to the Kubernetes default service, am I right?